src/compiler/utilities.ts

```diff
@@ -2769,6 +2769,14 @@ namespace ts {
         return heritageClause && heritageClause.types.length > 0 ? heritageClause.types[0] : undefined;
     }
 
+    export function getEffectiveImplementsTypeNodes(node: ClassLikeDeclaration): undefined | readonly ExpressionWithTypeArguments[]{

```

Comment:
it would be a lot safer to put this code into getClassImplementsHeritageClauseElements, and then change the name to `getEffectiveImplementsTypeNodes`.